.github/workflows/repro_check.yml

```
@@ -0,0 +1,79 @@
+name: Build and Diff Projects
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+      - reproducible
+
+jobs:
+  build_and_compare:
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-24.04
```
Let's pin the ubuntu version to avoid surprises